### PR TITLE
Universal Amazon Prime endpoints

### LINKED ISSUE #288: Amazon Prime fail to sync
### Expected Behavior

When I sync Amazon Prime fail, it will display the trakt data in right hand size, and allow user to tick the tick-box to sync

### Current Behavior

When the extension load the trakt data, it will not show anything and display "Failed to load Trakt history." 
The tick-box does not allow user tick

It seems cannot load the trakt data when sync Amazon Prime, but it doesn't have problem when sync Netflix.

1. Open the extension.
2. Choose Amazon Prime to sync
3. Wait 2-3 second it will display the error message.

It doesn't have problem when sync Netflix.
I tried remove Amazon Prime and Trakt cookies but it doesn't help.
After I delete the extension and install again, it will work 1 time, and will fail to load Trakt history after that

- **OS + Version: Windows 11 22H2
- **Browser + Version: Chrome 110.0.5481.178
- **Country/Region: Hong Kong

### PR DESCRIPTION
**Pull request by MrMamen**
- Fix prime sync for Prime Brazil
- Universal Region fix for Amazon Prime
Co-authored-by: Throne3d <Throne3d@users.noreply.github.com>

### CODE REVIEW

**Comment by MrMamen** [comment]
@rafaelgomesxyz Can't seem to understand why the linting fails. Works locally for me.

**Comment by MrMamen** [comment]
@Throne3d Please check if this still works in your region.

**Comment by Throne3d** [comment]
Looks to be working, yep! Notably, I did have to go to the "Account & Settings" page and re-authenticate first (https://www.primevideo.com/region/na/settings/ref=atv_nb_set) before it would work, even though I was logged in on the main primevideo.com site. Maybe we could list that as an extra troubleshooting step for Prime, alongside the change here?

File: src/services/amazon-prime/AmazonPrimeApi.ts [hunk 1]
```diff
@@ -264,7 +267,7 @@ class _AmazonPrimeApi extends ServiceApi {
 					const enrichments = enrichmentsResponse.enrichments[partialHistoryItem.id];
 					historyItems.push({
 						...partialHistoryItem,
-						progress: enrichments?.progress?.percentage ?? 0,
+						progress: enrichments?.progress?.percentage ?? 100,
```
Comment (rafaelgomesxyz) [inline src/services/amazon-prime/AmazonPrimeApi.ts:270]: Why are we defaulting to 100 when there's no percentage?

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-lockfileVersion: 5.3
+lockfileVersion: 5.4
```
Comment (rafaelgomesxyz) [inline pnpm-lock.yaml]: I've fixed this on master, so you can rebase and it should be good.

**Comment by rafaelgomesxyz** [comment]
So wait, I'm confused, how is this PR different than https://github.com/trakt-tools/universal-trakt-scrobbler/pull/274?

File: src/services/amazon-prime/AmazonPrimeApi.ts (hunk 1 above)
Comment (MrMamen) [reply src/services/amazon-prime/AmazonPrimeApi.ts:270]: At least at in my region, finished titles has no progress.

**Comment by MrMamen** [comment]
> So wait, I'm confused, how is this PR different than https://github.com/trakt-tools/universal-trakt-scrobbler/pull/274?

#274 is a WIP hardcoded for one region. This is a more general fix. I branches this off the other branch for history and credits.

**Comment by MrMamen** [comment]
Sorry. I see that I was working on the same branch as PR #274 so they are identical indeed.

**Comment by rafaelgomesxyz** [comment]
Oh I see, I'll add `Co-authored-by: Throne3d` when merging to give them credit.

But, is it actually working for you guys? I'm getting `{"error":{"errorCode":"PRS.InvalidRequest","message":"Cannot complete request.","type":"PRSInvalidRequestException"}}` in my responses from the requests to `ITEM_URL`, so history isn't loading and scrobbling is broken.

**Review by rafaelgomesxyz — APPROVED** [review]
Nevermind, seems to be an issue with the logic to set cookies when using Firefox + Containers. I'll fix in a separate PR. Approving.

Don't forget to add this to the end of the description when merging:

```
Co-authored-by: Throne3d <Throne3d@users.noreply.github.com>
```

**Comment by MrMamen** [comment]
I added the co-authored-tag, but is it necessary? Isn't this automatically added as long as it is at least one commit by another developer?

**Pull request merged**